airtime_mvc/application/common/FileDataHelper.php

```diff
@@ -68,4 +68,369 @@ public static function getFileExtensionFromMime($mime)
+    public static function resetArtwork($trackid)
+    {
+        $file = Application_Model_StoredFile::RecallById($trackid);
+        $md = $file->getMetadata();
+
+        $storDir = Application_Model_MusicDir::getStorDir();
+        $fp = $storDir->getDirectory();
+
+        $dbAudioPath = $md["MDATA_KEY_FILEPATH"];
+        $fullpath = $fp . $dbAudioPath;
+
+        $getID3 = new getID3;
+        $getFileInfo = $getID3->analyze($fullpath);
+
+        if(isset($getFileInfo['comments']['picture'][0])) {
+
+              $get_img = "";
+              $mime = $getFileInfo['comments']['picture'][0]['image_mime'];
+              $Image = 'data:'.$getFileInfo['comments']['picture'][0]['image_mime'].';charset=utf-8;base64,'.base64_encode($getFileInfo['comments']['picture'][0]['data']);
+              $base64 = @$Image;
+
+              $audioPath = dirname($fullpath);
+              $dbPath = dirname($dbAudioPath);
+              $normalizeValue = self::normalizePath($fullpath);
+              $path_parts = pathinfo($normalizeValue);
+              $file = $path_parts['filename'];
+
+              //Save Data URI
+              if (file_put_contents($audioPath . "/" . $file, $base64)) {
+                  $get_img = $dbPath . "/" . $file;
+              } else {
+                  Logging::error("Could not save Data URI");
+              }
+
+              $rfile = $audioPath . "/" . $file;
+
+              if ($mime == "image/png") {
+                  $ext = 'png';
+              } elseif ($mime == "image/gif") {
+                  $ext = 'gif';
+              } elseif ($mime == "image/bmp") {
+                 $ext = 'bmp';
+              } else {
+                 $ext = 'jpg';
+              }
+
+              if (file_exists($rfile)) {
+                  self::resizeImage($rfile, $rfile . '-32.jpg', $ext, 32, 100);
+                  self::resizeImage($rfile, $rfile . '-64.jpg', $ext, 64, 100);
+                  self::resizeImage($rfile, $rfile . '-128.jpg', $ext, 128, 100);
+                  self::resizeImage($rfile, $rfile . '-256.jpg', $ext, 256, 100);
+                  self::resizeImage($rfile, $rfile . '-512.jpg', $ext, 512, 100);
+                  self::resizeImage($rfile, $rfile . '-1024.jpg', $ext, 1024, 100);
+                  self::imgToDataURI($rfile . '-32.jpg', $rfile . '-32');
+                  self::imgToDataURI($rfile . '-64.jpg', $rfile . '-64');
+                  self::imgToDataURI($rfile . '-128.jpg', $rfile . '-128');
+                  self::imgToDataURI($rfile . '-256.jpg', $rfile . '-256');
+              } else {
+                  Logging::info("The file $rfile does not exist");
+              }
+        } else {
+              $get_img = "";
+        }
+        return $get_img;
+    }
+
+    /**
+     * Normalize string
+     *
+     * @param string $string
+     *
+     * @return string Normalized string
+     */
+    private static function normalizePath($string)
+    {
+        static $normal = array (
```

Comment:
Everything would work as it should. The only difference is that the file will be named without the character (eg. "Úsame.jpg" will then be "same.jpg") if we don't use normalize. I added it just to keep things more organized but can improved and even use the function to do the same for audio files since it currently saves files without those characters. So if we use it, "Marc Anthony Hits/same.mp3" would then be  "Marc_Anthony_Hits/Usame.mp3"

Reply:
This was just my way of organizing files. Sure we can possibly save as unicode and decode, but I prefer having them already named if it's going to be a physical file and is something that could also be used in audio files seeing how LB also allows you to Download a file. If trying to download a song named "Úsame.mp3" right now, you're just going to get a file named 'same.mp3', so it will need be saved as unicode there as well and decoded right before a download. I just prefer making things a bit cleaner this way.

Reply:
Sorry, my question was more this: If a file has a unicode char that is not on this list, what would break?

Reply:
> Sure we can possibly save as unicode and decode
What do you mean by this?

Reply:
Nothing would break. Once it saves into the db, that's what's going to be used to from now on.